core/plugins/org.polarsys.capella.core.semantic.queries/src/org/polarsys/capella/core/semantic/queries/basic/queries/ModelElementConstraints.java

```diff
@@ -54,6 +55,8 @@ public ModelElementConstraints() {
     List<Object> result = new ArrayList<Object>();
     if (object instanceof ModelElement) {
       EList<AbstractConstraint> constraints = ECollections.newBasicEList();
+      if (object instanceof Entity) 

```

Comment:
This is not the good fix. The original problem is rather on the creation of constraint. When you create a constraint and associate it to an Entity or an OA, it associates to the Entity/OA but not the part. That's why this query does not return anything. When you do the same thing but for a Logical Component for example, you'll see that the constraint is properly associated to the part and when you click on the Logical Component, you'll see the constraining elements.